src/main/java/net/schmizz/sshj/common/ECDSAVariationsAdapter.java

```diff
@@ -103,7 +103,7 @@ static void writePubKeyContentsIntoBuffer(PublicKey pk, Buffer<?> buf) {
     }
 
     static boolean isECKeyWithFieldSize(Key key, int fieldSize) {
-        return "ECDSA".equals(key.getAlgorithm())
+        return ("ECDSA".equals(key.getAlgorithm()) || "EC".equals(key.getAlgorithm()))
```

Comment:
Not sure where `"EC"` is coming from? Might want to add a test to cover that branch if that's possible.

Reply:
While you're changing it, also extract `"ECDSA"` into a constant, not sure why I didn't do that.

Reply:
The extraction should be done with the latest commit and it also contains a comment that explains where the "EC" comes from - it is the standard algorithm identifier for everything EC-related in the Android Keystore. Unfortunately, testing this seems impossible when not on the Android platform.